test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +91,9 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if (new Parser().parseCommand(inputCommand).isMutating()) {
+            assertEquals(addressBook, saveFile.load());
+        }
```

Comment:
These lines ensure that the scenario where the command is mutating is tested, but does not check the situations where non-mutating commands are issued instead.
Perhaps an extra else-statement could be added to test for these.